uniffi_macros/src/enum_.rs

```diff
@@ -202,3 +211,25 @@ pub fn variant_metadata(enum_: &DataEnum) -> syn::Result<Vec<TokenStream>> {
         )
         .collect()
 }
+
+/// Generate the `handle_callback_unexpected_error()` implementation
+///
+/// If handle_unknown_callback_error is true, this will use the `From<UnexpectedUniFFICallbackError>`
+/// implementation that the library author must provide.
+///
+/// If handle_unknown_callback_error is false, then we won't generate any code, falling back to the default
+/// implementation which panics.
+pub(crate) fn handle_callback_unexpected_error_fn(
+    handle_unknown_callback_error: bool,
+) -> TokenStream {
+    if handle_unknown_callback_error {
+        quote! {
+            fn handle_callback_unexpected_error(e: ::uniffi::UnexpectedUniFFICallbackError) -> Self {
+                <Self as ::std::convert::From<::uniffi::UnexpectedUniFFICallbackError>>::from(e)
+            }
+        }
+    } else {
+        // Use the default function
+        quote! {}
+    }
+}
```

Comment:
`Option<TokenStream>` can be interpolated the same way as `TokenStream`, so no code outside this function has to change if you use `bool::then` here:

```suggestion
) -> Option<TokenStream> {
    handle_unknown_callback_error.then(|| quote! {
        fn handle_callback_unexpected_error(e: ::uniffi::UnexpectedUniFFICallbackError) -> Self {
            <Self as ::std::convert::From<::uniffi::UnexpectedUniFFICallbackError>>::from(e)
        }
    })
}
```

(I think the comment in the else branch isn't really needed, the same thing is stated at the end of the doc comment)